Subject: [PATCH] disable self updates
---
Index: komelia-app/src/androidMain/kotlin/snd/komelia/Dependencies.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/komelia-app/src/androidMain/kotlin/snd/komelia/Dependencies.kt b/komelia-app/src/androidMain/kotlin/snd/komelia/Dependencies.kt
--- a/komelia-app/src/androidMain/kotlin/snd/komelia/Dependencies.kt	(revision aa2ba120ad9dc3ad95265e6230ed120884e07065)
+++ b/komelia-app/src/androidMain/kotlin/snd/komelia/Dependencies.kt	(date 1738508663569)
@@ -165,7 +165,6 @@
         .ktor(ktorWithCache)
         .build()
 
-    val appUpdater = createAppUpdater(ktorWithCache, ktorWithoutCache, context)
     return AndroidDependencyContainer(
         settingsRepository = settingsRepository,
         epubReaderSettingsRepository = epubReaderSettingsRepository,
@@ -176,7 +175,7 @@
         bookColorCorrectionRepository = bookColorCorrectionRepository,
         secretsRepository = secretsRepository,
 
-        appUpdater = appUpdater,
+        appUpdater = null,
         imageDecoderDescriptor = emptyFlow(),
         komgaClientFactory = komgaClientFactory,
         coilImageLoader = coil,
Index: komelia-app/src/androidMain/AndroidManifest.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/komelia-app/src/androidMain/AndroidManifest.xml b/komelia-app/src/androidMain/AndroidManifest.xml
--- a/komelia-app/src/androidMain/AndroidManifest.xml	(revision aa2ba120ad9dc3ad95265e6230ed120884e07065)
+++ b/komelia-app/src/androidMain/AndroidManifest.xml	(date 1738508733247)
@@ -2,7 +2,6 @@
 <manifest xmlns:android="http://schemas.android.com/apk/res/android">
 
     <uses-permission android:name="android.permission.INTERNET"/>
-    <uses-permission android:name="android.permission.REQUEST_INSTALL_PACKAGES"/>
 
     <application
             android:usesCleartextTraffic="true"
