Subject: [PATCH] disable self updates
---
Index: komelia-app/src/androidMain/kotlin/snd/komelia/AndroidDependencyContainer.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/komelia-app/src/androidMain/kotlin/snd/komelia/AndroidDependencyContainer.kt b/komelia-app/src/androidMain/kotlin/snd/komelia/AndroidDependencyContainer.kt
--- a/komelia-app/src/androidMain/kotlin/snd/komelia/AndroidDependencyContainer.kt	(revision 9ff7b73192ddcdf1731ab8e60f9a00dff73d3924)
+++ b/komelia-app/src/androidMain/kotlin/snd/komelia/AndroidDependencyContainer.kt	(date 1731157482634)
@@ -145,12 +145,11 @@
                 .ktor(ktorWithCache)
                 .build()
 
-            val appUpdater = createAppUpdater(ktorWithCache, ktorWithoutCache, context)
             return AndroidDependencyContainer(
                 settingsRepository = settingsRepository,
                 readerSettingsRepository = readerSettingsRepository,
                 secretsRepository = secretsRepository,
-                appUpdater = appUpdater,
+                appUpdater = null,
                 imageDecoderDescriptor = emptyFlow(),
                 komgaClientFactory = komgaClientFactory,
                 imageLoader = coil,
Index: komelia-app/src/androidMain/AndroidManifest.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/komelia-app/src/androidMain/AndroidManifest.xml b/komelia-app/src/androidMain/AndroidManifest.xml
--- a/komelia-app/src/androidMain/AndroidManifest.xml	(revision 9ff7b73192ddcdf1731ab8e60f9a00dff73d3924)
+++ b/komelia-app/src/androidMain/AndroidManifest.xml	(date 1731157482624)
@@ -2,7 +2,6 @@
 <manifest xmlns:android="http://schemas.android.com/apk/res/android">
 
     <uses-permission android:name="android.permission.INTERNET"/>
-    <uses-permission android:name="android.permission.REQUEST_INSTALL_PACKAGES"/>
 
     <application
             android:usesCleartextTraffic="true"
