diff --git a/node_modules/@d-i-t-a/reader/dist/esm/index.js b/node_modules/@d-i-t-a/reader/dist/esm/index.js
index bbea068..155e7bb 100644
--- a/node_modules/@d-i-t-a/reader/dist/esm/index.js
+++ b/node_modules/@d-i-t-a/reader/dist/esm/index.js
@@ -58931,8 +58931,8 @@ var _TouchEventHandler = class _TouchEventHandler {
               return;
             }
             const devicePixelRatio = window.devicePixelRatio;
-            const xDevicePixels = (startTouch.clientX - endTouch.clientX) / devicePixelRatio;
-            const yDevicePixels = (startTouch.clientY - endTouch.clientY) / devicePixelRatio;
+            const xDevicePixels = (startTouch.screenX - endTouch.screenX) / devicePixelRatio;
+            const yDevicePixels = (startTouch.screenY - endTouch.screenY) / devicePixelRatio;
             if (Math.abs(xDevicePixels) < _TouchEventHandler.TAP_TOLERANCE && Math.abs(yDevicePixels) < _TouchEventHandler.TAP_TOLERANCE) {
               if (touchEventEnd) {
                 touchEventStart = null;
@@ -58953,7 +58953,7 @@ var _TouchEventHandler = class _TouchEventHandler {
               touchEventStart = null;
               return;
             }
-            const slope = (startTouch.clientY - endTouch.clientY) / (startTouch.clientX - endTouch.clientX);
+            const slope = (startTouch.screenY - endTouch.screenY) / (startTouch.screenX - endTouch.screenX);
             if (Math.abs(slope) > 0.5) {
               touchEventStart = null;
               return;
